src/embed.rs

```diff
@@ -0,0 +1,268 @@
+    let mut embed = EmbedArea::for_file(&mut file)?;
+
+    embed.seek_to_start()?;
+    let mut buf = embed.new_buffer();
+    embed.read(&mut buf)?;
+    // compare to zeroed buffer
+    if buf == embed.new_buffer() {
+        bail!("No embedded Ignition config.");
+    }
+    let config =
+        String::from_utf8(extract_cpio(&buf)?).chain_err(|| "couldn't decode Ignition config")?;
+    print!("{}", config);
+    Ok(())
+}
+
+pub fn iso_remove(config: &IsoRemoveConfig) -> Result<()> {
+    let mut holder = CopiedFileHolder::new(&config.input, config.output.as_ref())?;
+    let mut embed = EmbedArea::for_file(&mut holder.file)?;
+    embed.seek_to_start()?;
+    let zeroes = embed.new_buffer();
+    embed.write(&zeroes)?;
+    holder.complete();
+    Ok(())
+}
+
+struct CopiedFileHolder {
+    pub file: File,
+    copied_path: Option<String>,
+    complete: bool,
+}
+
+/// Holder for a read/write file handle which is optionally copied from
+/// another file.  If complete() is not called and the file was copied,
+/// the copy will be deleted on drop.
+impl CopiedFileHolder {
+    fn new(input_path: &str, output_path: Option<&String>) -> Result<Self> {
+        if let Some(unwrapped_output_path) = output_path {
+            let mut input = OpenOptions::new()
+                .read(true)
+                .open(&input_path)
+                .chain_err(|| format!("opening {}", &input_path))?;
+            let mut output = OpenOptions::new()
+                .read(true)
+                .write(true)
+                .create_new(true)
+                .open(&unwrapped_output_path)
+                .chain_err(|| format!("opening {}", &unwrapped_output_path))?;
+            copy(&mut input, &mut output)
+                .chain_err(|| format!("copying {} to {}", input_path, unwrapped_output_path))?;
+            Ok(CopiedFileHolder {
+                file: output,
+                copied_path: Some(unwrapped_output_path.to_string()),
+                complete: false,
+            })
+        } else {
+            let file = OpenOptions::new()
+                .read(true)
+                .write(true)
+                .open(&input_path)
+                .chain_err(|| format!("opening {}", &input_path))?;
+            Ok(CopiedFileHolder {
+                file,
+                copied_path: None,
+                complete: false,
+            })
+        }
+    }
+
+    fn complete(&mut self) {
+        self.complete = true;
+    }
+}
+
+impl Drop for CopiedFileHolder {
+    fn drop(&mut self) {
+        if self.copied_path.is_some() && !self.complete {
+            let path = self.copied_path.as_ref().unwrap();
+            if let Err(e) = remove_file(path) {
+                eprintln!("Couldn't remove {}: {}", path, e);
+            }
+        }
+    }
+}
+
+struct EmbedArea<'a> {
+    file: &'a File,
+    offset: u64,
+    length: usize,
+}
+
+impl<'a> EmbedArea<'a> {
+    fn for_file(file: &'a mut File) -> Result<Self> {
```

Comment:
Can we document somewhere the metadata we are expecting?
From my reading of the code, we read the last 24B of this 32KiB area at the beginning of the disk and we expect the following:
 * `[u8; 8]` - magic "coreiso+"
 * `u64` little-endian - offset (from ISO byte 0?) of the cpio archive
 * `u64` little-endian - length of the cpio archive